lib/loopback.js

```diff
@@ -194,8 +216,197 @@ loopback.getCurrentContext = function() {
+};
+
+/**
+ * Look up a model class by name from all models created by
+ * `loopback.createModel()`. Throw an error when no such model exists.
+ *
+ * @param {String} modelName The model name
+ * @returns {Model} The model class
+ *
+ * @header loopback.getModel(modelName)
+ */
+loopback.getModel = function(modelName) {
+  return this.registry.getModel.apply(this.registry, arguments);
+};
+
+/**
+ * Look up a model class by the base model class.
+ * The method can be used by LoopBack
+ * to find configured models in models.json over the base model.
+ * @param {Model} modelType The base model class
+ * @returns {Model} The subclass if found or the base class
+ *
+ * @header loopback.getModelByType(modelType)
+ */
+loopback.getModelByType = function(modelType) {
+  return this.registry.getModelByType.apply(this.registry, arguments);
+};
+
+/**
+ * Create a data source with passing the provided options to the connector.
+ *
+ * @param {String} name Optional name.
+ * @options {Object} options Data Source options
+ * @property {Object} connector LoopBack connector.
+ * @property {*} [*] Other&nbsp;connector properties.
+ *   See the relevant connector documentation.
+ */
+
+loopback.createDataSource = function(name, options) {
+  return this.registry.createDataSource.apply(this.registry, arguments);
+};
+
+/**
+ * Get an in-memory data source. Use one if it already exists.
+ *
+ * @param {String} [name] The name of the data source.
+ * If not provided, the `'default'` is used.
+ */
+
+loopback.memory = function(name) {
+  return this.registry.memory.apply(this.registry, arguments);
+}
+
+/**
+ * Set the default `dataSource` for a given `type`.
+ * @param {String} type The datasource type.
+ * @param {Object|DataSource} dataSource The data source settings or instance
+ * @returns {DataSource} The data source instance.
+ *
+ * @header loopback.setDefaultDataSourceForType(type, dataSource)
+ */
+
+loopback.setDefaultDataSourceForType = function(type, dataSource) {
+  return this.registry.setDefaultDataSourceForType.apply(this.registry, arguments);
+};
+
+/**
+ * Get the default `dataSource` for a given `type`.
+ * @param {String} type The datasource type.
+ * @returns {DataSource} The data source instance
+ */
+
+loopback.getDefaultDataSourceForType = function(type) {
+  return this.registry.getDefaultDataSourceForType.apply(this.registry, arguments);
+};
+
+/**
+ * Attach any model that does not have a dataSource to
+ * the default dataSource for the type the Model requests
+ */
+
+loopback.autoAttach = function() {
+  return this.registry.autoAttach.apply(this.registry, arguments);
+};
+
+loopback.autoAttachModel = function(ModelCtor) {
+  return this.registry.autoAttachModel.apply(this.registry, arguments);
+};
+
+// temporary alias to simplify migration of code based on <=2.0.0-beta3
+Object.defineProperty(loopback, 'DataModel', {
+  get: function() {
+    return this.registry.DataModel;
+  }
+});
```

Comment:
Is there any particular reason for writing these methods by hand, instead of performing an automatic mixin?

```js
Object.keys(loopback.registry).forEach(function(k) {
  var target = loopback.registry[key];
  if (typeof target !== 'function') return;
  if (loopback[key]) return;
  loopback[key] = function() { this.registry[key].apply(this.registry, arguments); };
});
```